assembl/static2/js/app/components/debate/thread/answerForm.jsx

```diff
@@ -1,36 +1,102 @@
 import React from 'react';
-import { Row, Col, FormGroup, FormControl, Button } from 'react-bootstrap';
+import { PropTypes } from 'prop-types';
+import { connect } from 'react-redux';
+import { graphql } from 'react-apollo';
+import { Row, Col, FormGroup, Button } from 'react-bootstrap';
 import { Translate, I18n } from 'react-redux-i18n';
 
+import createPostMutation from '../../../graphql/mutations/createPost.graphql';
+import {
+  updateAnswerPostBody,
+  updateAnswerPostBodyRemaingChars,
+  updateAnswerPostFormStatus
+} from '../../../actions/postsActions';
+import { displayModal, displayAlert } from '../../../utils/utilityManager';
+import { getConnectedUserId } from '../../../utils/globalFunctions';
+import { getCurrentView, getContextual } from '../../../utils/routeMap';
+import { TxtAreaWithRemainingChars } from '../../common/txtAreaWithRemainingChars';
+
 const TEXT_AREA_MAX_LENGTH = 3000;
 const TEXT_AREA_ROWS = 10;
 
-const AnswerForm = ({ body }) => {
-  const handleBodyChange = () => {};
-  const resetForm = () => {};
-  const answerPost = () => {};
+const AnswerForm = ({
+  body,
+  bodyRemainingChars,
+  mutate,
+  updateBody,
+  updateBodyChars,
+  slug,
+  ideaId,
+  parentId,
+  refetchIdea,
+  updateAnswerFormStatus
+}) => {
+  const handleBodyChange = (e) => {
+    const maxChars = TEXT_AREA_MAX_LENGTH;
+    const length = e.target.value.length;
+    const remaining = maxChars - length;
+    updateBodyChars(remaining);
+    updateBody(e.target.value);
+  };
+
+  const resetForm = () => {
+    updateAnswerFormStatus(false);
+    updateBodyChars(TEXT_AREA_MAX_LENGTH);
+    updateBody('');
+  };
+
+  const inviteToLogin = () => {
```

Comment:
inviteToLogin function is duplicated between answerForm.jsx and topPostForm.jsx. I guess it should be imported from a separate file (and optionnaly call it with configuration parameters).